include/uxr/agent/Agent.hpp

```diff
@@ -0,0 +1,341 @@
+     * @param client_key        The identifier of the ProxyClient.
+     * @param datawriter_id     The identifier of the DataWriter to be created.
+     * @param publisher_id      The identifier of the Publisher associated with the DataWriter.
+     * @param xml               The XML that describes the DataWriter.
+     * @param flag              The flag that determines the creation mode.
+     * @param errcode           The result status of the operation.
+     * @return  true in case of success and false in other case.
+     */
+    microxrcedds_agent_DllAPI static bool create_datawriter_by_xml(
+            uint32_t client_key,
+            uint16_t datawriter_id,
+            uint16_t publisher_id,
+            const char* xml,
+            uint8_t flag,
+            ResultStatus& errcode);
+
+    /**
+     * @brief Creates a DataReader from a reference in the ProxyClient identified by the client_key.
+     *        The DataReader is associated with a Subscriber identified by subscriber_id.
+     * @param client_key        The identifier of the ProxyClient.
+     * @param datareader_id     The identifier of the DataReader to be created.
+     * @param subscriber_id     The identifier of the Subscriber associated with the DataReader.
+     * @param ref               The reference of the DataReader.
+     * @param flag              The flag that determines the creation mode.
+     * @param errcode           The result status of the operation.
+     * @return  true in case of success and false in other case.
+     */
+    microxrcedds_agent_DllAPI static bool create_datareader_by_ref(
+            uint32_t client_key,
+            uint16_t datareader_id,
+            uint16_t subscriber_id,
+            const char* ref,
+            uint8_t flag,
+            ResultStatus& errcode);
+
+    /**
+     * @brief Creates a DataReader from an XML in the ProxyClient identified by the client_key.
+     *        The DataReader is associated with a Subscriber identified by subscriber_id.
+     * @param client_key        The identifier of the ProxyClient.
+     * @param datareader_id     The identifier of the DataReader to be created.
+     * @param subscriber_id     The identifier of the Subscriber associated with the DataReader.
+     * @param xml               The XML that describes the DataReader.
+     * @param flag              The flag that determines the creation mode.
+     * @param errcode           The result status of the operation.
+     * @return  true in case of success and false in other case.
+     */
+    microxrcedds_agent_DllAPI static bool create_datareader_by_xml(
+            uint32_t client_key,
+            uint16_t datareader_id,
+            uint16_t subscriber_id,
+            const char* xml,
+            uint8_t flag,
+            ResultStatus& errcode);
+
+    /**
+     * @brief Deletes an entity identifier by the object_id in a ProxyClient idenfied by the client_key.
```

Comment:
```suggestion
     * @brief Deletes an entity identified by the object_id in a ProxyClient identified by the client_key.
```

Reply:
Also, I think that it is interesting to comment that if you delete an entity, all its associated entities are removed too (because this is the behaviour right?)

Reply:
Ok, fixed!